mesonbuild/modules/external_project.py

```diff
@@ -93,7 +93,7 @@ def __init__(self,
         # will install files into "c:/bar/c:/foo" which is an invalid path.
         # Work around that issue by removing the drive from prefix.
         if self.prefix.drive:
-            self.prefix = self.prefix.relative_to(self.prefix.drive)
+            self.prefix = Path('/' + str(self.prefix.relative_to(self.prefix.drive + "/")))
```

Comment:
The problem here is actually that `Path.relative_to()` is horrible and must not be used under any circumstances. We've run into this issue before: https://github.com/mesonbuild/meson/pull/12998#pullrequestreview-1947396217

Simply avoid it like the plague (and in general avoid pathlib for anything that involves concrete paths), and rewrite this code to use `mesonlib.relpath`.

Reply:
Done at 9f2acb2

Reply:
Please use `git commit --fixup=<sha1> && git rebase -i --autosquash` or similar, in order to fix up older commits -- rather than adding additional commits to revert a change and make a different change.

Each commit should represent one logically distinctive change, such as:
- making external_project work on msys2
- make the testsuite handle msys2's addition of `cmd`